sway/desktop/xwayland.c

```diff
@@ -287,8 +287,6 @@ static void handle_commit(struct wl_listener *listener, void *data) {
 	if (view->swayc->instruction) {
 		transaction_notify_view_ready_by_size(view,
 				surface_state->width, surface_state->height);
-	} else if (container_is_floating(view->swayc)) {
```

Comment:
Hmm, is there any bad consequence of removing this?

Maybe we should have common code for all shells doing that?

Reply:
Your commit that I cherry picked removed the `view_set_{position,size}` functions. Maybe it wasn't called here at the time you made your commit.

I'm also unsure if there's any bad consequences of removing it. Maybe we could leave it out and see. If anything does need it, we can put it back and add a comment saying which program needs it.

Reply:
I now remember what the issue is:
1. Open Firefox
2. Initiate a file download (e.g. go to https://www.debian.org/distrib/ and click on "64-bit PC netinst iso")
3. The "Open file/save as" dialog appears. It's initially small and is then resized to a larger size. It's broken with this patch.